### PR TITLE
Added Strafe

### PR DESCRIPTION
**Pull request by FluorideHistorian**
**Describe the pull**
Added fast strafe which bypasses ncp

**Describe how this pull is helpful**
#633 #180 #852 👀 

**Additional context**
Code is kinda bad but it works

### CODE REVIEW

File: src/main/java/me/zeroeightsix/kami/util/StrafeUtils.kt [hunk 1]
```diff
@@ -0,0 +1,19 @@
+package me.zeroeightsix.kami.util
+
+import net.minecraft.client.Minecraft
+import kotlin.math.*
+
+class StrafeUtils {
```
Comment (Xiaro) [inline src/main/java/me/zeroeightsix/kami/util/StrafeUtils.kt:6]: You can make this as an object instead so you don't have to create a new instance every time you use it in a class

File: src/main/java/me/zeroeightsix/kami/util/StrafeUtils.kt [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+    fun getDirection(): Double {
+        return Math.toRadians(Minecraft.getMinecraft().player.rotationYaw.toDouble())
```
Comment (Xiaro) [inline src/main/java/me/zeroeightsix/kami/util/StrafeUtils.kt:8]: You can replace all `Minecraft.getMinecraft()` with a constant

**Comment by FluorideHistorian** [comment]
oh yeah also #935

File: src/main/java/me/zeroeightsix/kami/module/modules/movement/Strafe.kt [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+
+    private var jumpTicks = 0
+    private val strafeUtils = StrafeUtils()
+
+    /* if you skid this you omega gay */
+    override fun onUpdate() {
+        if(mc.gameSettings.keyBindForward.isKeyDown) {
+            strafeUtils.setSpeed(strafeUtils.getSpeed())
+            if(airSpeedBoost.value)mc.player.jumpMovementFactor = 0.029F
+            if(timerBoost.value)mc.timer.tickLength = 50 / 1.09F
```
Comment (Xiaro) [inline src/main/java/me/zeroeightsix/kami/module/modules/movement/Strafe.kt:28]: I think having timer on for long time is what causing it not bypassing well. Maybe reset it to default at a point and turn it back on?

File: src/main/java/me/zeroeightsix/kami/util/StrafeUtils.kt (hunk 1 above)
Comment (FluorideHistorian) [reply src/main/java/me/zeroeightsix/kami/util/StrafeUtils.kt:6]: 😳

**Comment by iBuyMountainDew** [comment]
@FluorideHistorian have you tested it in practice in a variety of situations?

**Comment by FluorideHistorian** [comment]
@iBuyMountainDew not really just general ncp testing on multiple servers

**Comment by FluorideHistorian** [comment]
cant test right now because mc is being smooth brain and keeps crashing

**Comment by dominikaaaa** [comment]
> @FluorideHistorian have you tested it in practice in a variety of situations?

That doesn't matter at all, we should be doing testing ourselves

**Comment by FluorideHistorian** [comment]
@itsokk i cannot figure out how to do that at the moment and dont feel like skidding

**Comment by itsokk** [comment]
> 
> 
> @itsokk i cannot figure out how to do that at the moment and dont feel like skidding

add a comment that you have taken something from the client in the code, make sure its open source and the license allows, easy

**Comment by dominikaaaa** [comment]
Yea that's not skidding but it's not very easy to do

**Comment by FriedrichM** [comment]
make sure to check if it breaks elytra fly / takeoff

**Comment by dominikaaaa** [comment]
Elytraflight takeoff should just be a jump. I need to add a return to make sure it doesn't activate when elytra flying

**Comment by FluorideHistorian** [comment]
strafe works with elytra fly in my testing but still would be better to disable strafe while elytra flying

**Comment by FluorideHistorian** [comment]
strafe go brrrrrrrrrrrrr

**Pull request merged**